### PR TITLE
Fix loading scheduler when having more than one param_group

### PR DESCRIPTION
**Pull request by TJ-Solergibert**
Solves #221 

When training with PP every PP rank gets a different set & quantity of layers, so we need to store the `lr_scheduler` state dict of each PP rank to be able to resume training properly.  (Check size of state dicts)

image

Similar to how optimizer states and model parameters are stored, we just save the LR scheduler in the DP=TP=EP=0 ranks.

PD: The photo is from a first fix where accidentally stored a statedict for each TP rank (Useless)

### CODE REVIEW

File: src/nanotron/serialize/optimizer.py [hunk 1]
```diff
@@ -30,9 +30,9 @@ def optimizer_filename(parallel_context: ParallelContext, is_zero: bool):
         return f"{ObjectType.OPTIMIZER.value}_pp-{dist.get_rank(parallel_context.pp_pg)}-of-{parallel_context.pp_pg.size()}_tp-{dist.get_rank(parallel_context.tp_pg)}-of-{parallel_context.tp_pg.size()}_exp-{dist.get_rank(parallel_context.expert_pg)}-of-{parallel_context.expert_parallel_size}.pt"
 
 
-def lr_scheduler_filename():
+def lr_scheduler_filename(parallel_context: ParallelContext):
     """The lr_scheduler is the same for all processes."""
-    return f"{ObjectType.LR_SCHEDULER.value}.pt"
+    return f"{ObjectType.LR_SCHEDULER.value}_pp-{dist.get_rank(parallel_context.pp_pg)}-of-{parallel_context.pp_pg.size()}.pt"
```
Comment (NouamaneTazi) [inline src/nanotron/serialize/optimizer.py:35]: still need to fix this when different TPs have different number of lr_lambdas aka different number of params
So ideally we want the lr_scheduler to follow the same naming as optimizer (we can maybe make them share the same filename function)

**Review by NouamaneTazi — CHANGES_REQUESTED** [review]

File: src/nanotron/serialize/optimizer.py (hunk 1 above)
Comment (TJ-Solergibert) [reply src/nanotron/serialize/optimizer.py]: Now we use the same naming convention as the optimizer. Added `is_zero` flag which is obtained either from `config.optimizer.zero_stage` OR `optimizer.inherit_from(optim.ZeroDistributedOptimizer)`

**Review by NouamaneTazi — APPROVED** [review]
Great job! Tyy
